elytron/src/main/java/org/wildfly/extension/elytron/FileSystemRealmDefinition.java

```diff
@@ -425,4 +442,62 @@ private static Service getFileSystemService(OperationContext context) throws Ope
         return serviceContainer.getService();
     }
 
+    /**
+     * Integrity keypair can only be added at initialization, unless realm contains no identities. Existing realms should
+     * be upgraded with Elytron Tool.
+     *
+     * @see <a href="https://issues.redhat.com/browse/WFCORE-6129">WFCORE-6129</a>
+     */
+    private static class IntegrityWriteAttributeDisabledHandler extends ReloadRequiredWriteAttributeHandler {
+        public IntegrityWriteAttributeDisabledHandler(final AttributeDefinition... definitions) {
+            super(definitions);
+        }
+
+        @Override
+        protected boolean applyUpdateToRuntime(OperationContext context, ModelNode operation, String attributeName,
+                                               ModelNode resolvedValue, ModelNode currentValue, HandbackHolder<Void> voidHandback) throws OperationFailedException {
+
+            TrivialService<FileSystemSecurityRealm> filesystemService = (TrivialService<FileSystemSecurityRealm>) getFileSystemService(context);
+            FileSystemSecurityRealm fileSystemRealm = filesystemService.getValue();
+
+            try {
+                if (!currentValue.isDefined() && fileSystemRealm.getRealmIdentityIterator().hasNext()) {
+                    throw ROOT_LOGGER.addKeypairToInitializedFilesystemRealm();
+                }
+            } catch (RealmUnavailableException ignored) {
+                throw ROOT_LOGGER.addKeypairToInitializedFilesystemRealm();
+            }
+            return super.applyUpdateToRuntime(context, operation, attributeName, resolvedValue, currentValue, voidHandback);
+        }
+    }
+
+    /**
+     * Encryption secret key can only be added at initialization, unless realm contains no identities. Existing realms should
+     * be upgraded with Elytron Tool.
+     *
+     * @see <a href="https://issues.redhat.com/browse/JBEAP-24206">JBEAP-24206</a>
```

Comment:
@cam-rod Just a minor, we should reference WFCORE jira issue here as well

Reply:
thanks, missed this one. fixed